Roche-Engine/PlayerMovement.cpp

```diff
@@ -0,0 +1,61 @@
+#include "stdafx.h"
+#include "PlayerMovement.h"
+
+PlayerMovement::PlayerMovement(Physics* physics, float speed)
+{
+	m_physics = physics;
```

Comment:
I think changing the physics component to a shared_ptr might make more sense. You'd maintain the type from all the other classes and this would also remove the need to call ```delete``` on it as well.